ports/gstplugin/servosrc.rs

```diff
@@ -563,35 +535,80 @@ impl BaseSrcImpl for ServoSrc {
         _length: u32,
         buffer: &mut BufferRef,
     ) -> Result<FlowSuccess, FlowError> {
-        let guard = self.info.lock().map_err(|_| {
-            gst_element_error!(src, CoreError::Negotiation, ["Lock poisoned"]);
-            FlowError::NotNegotiated
-        })?;
-        let info = guard.as_ref().ok_or_else(|| {
-            gst_element_error!(src, CoreError::Negotiation, ["Caps not set yet"]);
-            FlowError::NotNegotiated
+        let memory = buffer.get_all_memory().ok_or_else(|| {
+            gst_element_error!(src, CoreError::Failed, ["Failed to get memory"]);
+            FlowError::Error
         })?;
-        let mut frame = VideoFrameRef::from_buffer_ref_writable(buffer, info).ok_or_else(|| {
-            gst_element_error!(
-                src,
-                CoreError::Failed,
-                ["Failed to map output buffer writable"]
-            );
+        let memory = unsafe { memory.into_ptr() };
```

Comment:
Regarding https://github.com/servo/servo/issues/24831#issuecomment-561365449 did you finally try with `as_ptr`?

Reply:
I've been focusing on the issue with buffer allocation rather than the space leak atm. I got buffer allocation to work over in my playground plugin, I'm going to port it here, and look to see if this makes the space leak magically go away. https://github.com/asajeffrey/my-gst-plugin/commit/14f6657066a8e4d2c3661c49c9c5482baad1d58c